benchmarks/performance_test.py

```diff
@@ -0,0 +1,505 @@
+
+    def single_loop(self, conditions, generate=False, area_def=None, resampler_kwargs=None):
+        """Single round of the test."""
+        import dask.config
+        self.running = True
+
+        chunk_size, num_worker, resampler = conditions
+
+        dask.config.set({"array.chunk-size": f"{chunk_size}MiB"})
+        dask.config.set(num_workers=num_worker)
+
+        try:
+            num_thread = os.environ["OMP_NUM_THREADS"]
+        except KeyError:
+            num_thread = psutil.cpu_count(logical=True)
+
+        print(f"Start testing CHUNK_SIZE={chunk_size}MiB, NUM_WORKER={num_worker}, NUM_THREADS={num_thread}, "  # noqa
+              f"""resampler is "{resampler}".""")
+
+        # Start recording cpu/mem usage
+        monitor_thread = Thread(target=self.monitor_system_usage, args=(0.5,))
+        monitor_thread.start()
+
+        errors = []
+        start = time.perf_counter()
+        try:
+            self.satpy_test(resampler, generate, area_def, resampler_kwargs)
+        except Exception as e:
+            errors.append(e)
+
+        end = time.perf_counter()
+        # All of these must be list object
+        self.result["process_time"] = [end - start]
+        self.result["scenes"] = [len(self.folders)]
+        self.result["errors"] = errors
+
+        # Stop recording
+        self.running = False
+        monitor_thread.join()
+
+        if area_def is None:
+            area = "original"
+        else:
+            area = "local" if len(area_def.area_id) == 0 else area_def.area_id.replace("_", "")
+
+        csv_file = (f"{self.work_dir}/{self.reader_name.replace('_', '')}_"
+                    f"chunk{chunk_size}_worker{num_worker}_thread{num_thread}_{area}_{resampler}.csv")
+        self.write_to_csv(csv_file)
+
+    def simple_test(self, diff_res=False):
+        """Test the reader in dataset's original projection, with no resampling or the simplest ``native`` resampling.
+
+        Args:
+            diff_res (bool): If the composite requires bands in different resolutions, this should be set to `True`
+                             so the native resampler will match them to the ``scn.finest_area()``.
+                             For example, ``true_color`` of ABI needs 500m C01 and 1000m C02 bands, so it's `True`.
+                             **This is not a test option and should be set properly according to the composite,**
+                             otherwise the test will end up in errors.
+
+        """
+        resampler = "native" if diff_res else "none"
+        generate = not diff_res
+        total_rounds = len(self.chunk_size_opts) * len(self.worker_opts)
+
+        print(f"{self.reader_name} test started. Composite is \"{self.composite}\".\n")  # noqa
+        i = 0
+        for chunk_size in self.chunk_size_opts:
+            for num_worker in self.worker_opts:
+                self.single_loop((chunk_size, num_worker, resampler), generate=generate)
+                i = i + 1
```

Comment:
```suggestion
for chunk_size, num_worker in itertools.product(self.chunk_size_opts, self.worker_opts):
```

I wonder if the final report could be put after the for loop and instead find a way to only do the sleep (maybe split the completed message and the 1-min rest message into two prints) if we aren't on the last round.